pkg/app/AppListingService.go

```diff
@@ -770,6 +798,51 @@ func (impl AppListingServiceImpl) FetchAppStageStatus(appId int, appType int) ([
 	return appStageStatuses, err
 }
 
+func (impl AppListingServiceImpl) fetchCiArtifactAndGitTriggersMap(artifactIds []int) (CiArtifactAndGitCommitsMap map[int][]string, err error) {
+	artifacts, err := impl.ciArtifactRepository.GetByIds(artifactIds)
```

Comment:
Can the length of the artifactIds array be zero? If yes, GetByIds func will return error. Handle this inside ciArtifactRepository.GetByIds by putting check on length of array